src/transformers/generation_utils.py

```diff
@@ -747,6 +840,16 @@ def greedy_search(
 
             >>> print("Generated:", tokenizer.batch_decode(outputs, skip_special_tokens=True))
         """
+        logits = None
+        decoder_attentions = () if output_attentions else None
+        decoder_hidden_states = None
+
+        if "encoder_outputs" in model_kwargs:
+            encoder_outputs = model_kwargs["encoder_outputs"]
+            encoder_attentions = encoder_outputs.get("encoder_attentions", None) if output_attentions else None
+            encoder_hidden_states = (
+                encoder_outputs.get("encoder_hidden_states", None) if output_hidden_states else None
+            )
 
         # init values
         logits_processor = logits_processor if logits_processor is not None else LogitsProcessorList()
```

Comment:
Actually can we move all those `# init values` lines over the added code? and add

`output_scores = output_scores if output_scores is not None else self.config.output_scores`
etc...